lib/project_types/extension/features/argo_serve_options.rb

```diff
@@ -0,0 +1,37 @@
+module Extension
+  module Features
+    class ArgoServeOptions
+      include SmartProperties
+      property! :context, accepts: ShopifyCli::Context
+      property  :port, accepts: Integer, default: 39351
+      property  :public_url, accepts: String, default: ""
+      property! :required_fields, accepts: Array, default: -> { [] }
+      property! :renderer_package, accepts: Features::ArgoRendererPackage
+
+      YARN_SERVE_COMMAND = %w(server)
+      NPM_SERVE_COMMAND = %w(run-script server)
+
+      def yarn_serve_command
+        YARN_SERVE_COMMAND + options
+      end
+
+      def npm_serve_command
+        NPM_SERVE_COMMAND  + ["--"] + options
+      end
+
+      private
+
+      def options
+        project = ExtensionProject.current
+        @serve_options ||= [].tap do |options|
+          options << "--port=#{port}"
+          options << "--shop=#{project.env.shop}" if required_fields.include?(:shop)
+          options << "--apiKey=#{project.env.api_key}" if required_fields.include?(:api_key)
+          options << "--argoVersion=#{renderer_package.version}" if renderer_package.admin?
+          options << "--uuid=#{project.registration_uuid}" if renderer_package.supports_uuid_flag?
```

Comment:
I think we should be consistent about passing these params as they are all tied to the argo-admin-cli version:
- `publicUrl`
- `uuid`
- `argoVersion`

Reply:
```
--entry="index.tsx" --port=39351 --type=PRODUCT_SUBSCRIPTION "--port=39352 [...]"
```

The Port is currently passed twice as it is also part of the template. The second port should take precedence and differs from the port specified in the template by 1.

Reply:
This is the PR Cecy is referring to: https://github.com/Shopify/shopify-app-cli/pull/1212

Reply:
Oh right, I understand now 👍